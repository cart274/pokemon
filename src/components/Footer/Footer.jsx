import React from "react";
import style from "./Footer.module.css";

const Footer = ({t = ()=>{}}) => {

  return (
    <section className={style.footer}>
      <h3>{t('Desarrollado por')} Carlos Rodr√≠guez</h3>
    </section>
  );
}

export default Footer;
