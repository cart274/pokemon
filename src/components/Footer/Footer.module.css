@value colors: "../../assets/styles/variables.css";
@value black, cerulean, atlantis, white, grey from colors;

.footer{
    padding: 80px 5% 40px 5%;
    text-align: center;
    background-image: url('../../assets/images/backgrounds/Bg_Footer.png')
}
.footer h2{
    text-align: center;
    font-size: 46px;
    font-weight: 600;
    margin-bottom: 10px;
}
.footer h2 span{
    color: cerulean;
    font-weight: 600;
}
.footer h3{
    text-align: center;
    font-size: 34px;
    font-weight: 500;
    margin-bottom: 40px;
}
.footer button{
    margin: 0 auto 120px auto;
}
.footer img{
    text-align: center;
}