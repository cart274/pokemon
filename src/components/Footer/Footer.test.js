import React from 'react';
import ReactDOM from 'react-dom';
import Footer from './Footer';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<Footer />', () => {

    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Footer />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshots', () => {
        const wrapper = shallow(<Footer/>);
        expect(wrapper).toMatchSnapshot();
    });
})
