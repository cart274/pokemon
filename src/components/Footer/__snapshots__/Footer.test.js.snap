// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3>
         Carlos Rodríguez
      </h3>,
      "className": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " Carlos Rodríguez",
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " Carlos Rodríguez",
      ],
      "type": "h3",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
           Carlos Rodríguez
        </h3>,
        "className": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " Carlos Rodríguez",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " Carlos Rodríguez",
        ],
        "type": "h3",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
